src/main/java/de/uni_tuebingen/ub/nppm/db/FrontendExtendedSearch.java

```diff
@@ -11,29 +11,50 @@
 public class FrontendExtendedSearch extends AbstractBase {
 
     public static List<Map<String, String>> getSearchResult(String fieldsString, String tablesString, String conditionsString, String order, String[] fields) throws Exception {
-        String sql = "SELECT DISTINCT "+fieldsString+" FROM "+tablesString+" WHERE ("+conditionsString+") "+order;
-        try (Session session = getSession()) {
-            NativeQuery sqlQuery = session.createNativeQuery(sql);
-            List<Object[]> rows = sqlQuery.getResultList();
-            //return var
-            List<Map<String, String>> ret = new ArrayList<Map<String, String>>();
-            //loop over the rows
-            for (Object[] row : rows) {
-                //convert the fields from the row to a map
-                Map<String, String> fieldVal = new HashMap<String, String>();
-                for (int i = 0; i < fields.length; i++) {
-                    String[] name = fields[i].split(" AS ");
-                    if (name.length == 2) {
-                        fields[i] = name[1];
+        String sql = "SELECT DISTINCT " + fieldsString + " FROM " + tablesString + " WHERE (" + conditionsString + ") " + order;
+        try ( Session session = getSession()) {
+            if (fields.length > 1) {
+                NativeQuery sqlQuery = session.createNativeQuery(sql);
+                List<Object[]> rows = sqlQuery.getResultList();
+                //return var
+                List<Map<String, String>> ret = new ArrayList<Map<String, String>>();
+                //loop over the rows
+                for (Object[] row : rows) {
+                    //convert the fields from the row to a map
+                    Map<String, String> fieldVal = new HashMap<String, String>();
+                    for (int i = 0; i < fields.length; i++) {
+                        String[] name = fields[i].split(" AS ");
+                        if (name.length == 2) {
+                            fields[i] = name[1];
+                        }
+                        if (row[i] != null) {
+                            fieldVal.put(fields[i].trim(), row[i].toString());
+                        }
                     }
-                    if (row[i] != null) {
-                        fieldVal.put(fields[i].trim(), row[i].toString());
+                    ret.add(fieldVal);
+                }//end for (Object[] row : rows)
+
+                return ret;
+            }//end fields.length
+            else {
+                NativeQuery sqlQuery = session.createNativeQuery(sql);
```

Comment:
This line as well as the initialization of the ret variable are the same in both blocks, so you might actually move them before the if...else